server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryFactory.java

```diff
@@ -288,31 +281,31 @@ private boolean mergeDeprecatedComponentParameters(DbSession session, Boolean on
       allComponentUuids.addAll(componentRootUuids);
       effectiveOnComponentOnly = false;
     } else if (componentRoots != null) {
-      allComponentUuids.addAll(componentUuids(session, componentRoots));
+      allComponentUuids.addAll(convertComponentKeysToUuids(session, componentRoots));
       effectiveOnComponentOnly = false;
     } else if (components != null) {
-      allComponentUuids.addAll(componentUuids(session, components));
+      allComponentUuids.addAll(convertComponentKeysToUuids(session, components));
       effectiveOnComponentOnly = true;
     } else if (componentUuids != null) {
       allComponentUuids.addAll(componentUuids);
       effectiveOnComponentOnly = BooleanUtils.isTrue(onComponentOnly);
     } else if (componentKeys != null) {
-      allComponentUuids.addAll(componentUuids(session, componentKeys));
+      allComponentUuids.addAll(convertComponentKeysToUuids(session, componentKeys));
       effectiveOnComponentOnly = BooleanUtils.isTrue(onComponentOnly);
     }
     return effectiveOnComponentOnly;
   }
 
   private static boolean atMostOneNonNullElement(Object... objects) {
-    return !from(Arrays.asList(objects))
-      .filter(notNull())
-      .anyMatch(new HasTwoOrMoreElements());
+    return Arrays.stream(objects)
+      .filter(Objects::nonNull)
+      .count() <= 1;
   }
 
   private void addComponentParameters(IssueQuery.Builder builder, DbSession session,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Method has 10 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)